src/app/patient-dashboard/common/todays-vitals/test-todays-vitals.service.ts

```diff
@@ -0,0 +1,94 @@
+import { Injectable } from '@angular/core';
+
+import { TodaysVitalsService } from './todays-vitals.service';
+import { VitalsDatasource } from './vitals.datasource';
+
+@Injectable()
+/**
+ * TestTodaysVitalsService pretends to make real http requests.
+ * implements only as much of TodaysVitalsService as is actually consumed by the app
+*/
+export class TestTodaysVitalsService extends TodaysVitalsService {
+
+  constructor(_vitalsDataSource: VitalsDatasource) {
```

Comment:
Avoid the use of underscore prefix for members. Any justification for this?
```suggestion
  constructor(vitalsDataSource: VitalsDatasource) {
```

Reply:
The constructor required that I have a different vitalsDataSource singleton than the one in the parent class I'd extended. Prefixing this private instance of vitalsDataSource with an underscore differentiated it with the one in the parent class and therefore resolved the error. More recently however, getting rid of TestTodaysVitalsService and replacing it with a simple stub resolves this.